includes/db.inc

```diff
@@ -308,6 +311,24 @@ function islandora_solr_get_collection_sort_string($pid, $use_fallbacks = FALSE)
 }
 
 /**
+ * Sets a default of ' asc' if no sort order string is provided.
+ *
+ * @param string $sort_setting
+ *   The sort setting value to apply the default to.
+ */
+function islandora_solr_set_default_collection_sort_order($sort_setting) {
+  $exploded_sort_setting = explode(',', $sort_setting);
+  foreach ($exploded_sort_setting as &$field) {
+    $field = trim($field);
+    if (!preg_match('/\s+(asc|desc)$/', $field)) {
```

Comment:
I'm uncertain (and don't have time right now to test for examples) but I think it may be possible to trip this up due to how very permissive the Solr syntax can be. For instance I think `foo asc` is a valid Solr field name in some versions.